test/test_unbalanced.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for module Unbalanced OT with entropy regularization"""
+
+# Author: Hicham Janati <hicham.janati@inria.fr>
+#
+# License: MIT License
+
+import numpy as np
+import ot
+import pytest
+
+
+@pytest.mark.parametrize("method", ["sinkhorn"])
+def test_unbalanced_convergence(method):
```

Comment:
That's not what I was suggesting. I was suggesting to test only the type alone and make sure that whenever is not `'sinkhorn'` it does whatever you expected it to do (i.e: raise an error).

Another way to write it without parametrization is as follows:

```py
def test_implemented_methods():
    IMPLEMENTED_METHODS = ['sinkhorn']
    TO_BE_IMPLEMENTED_METHODS = ['sinkhorn_stabilized', 'sinkhorn_epsilon_scaling']
    NOT_VALID_TOOKENS = ['foo', 'bar']
    
    for method in IMPLEMENTED_METHODS:
    	_ = ot.unbalanced.sinkhorn(default, default, method=method, None, whatever)
    
    with pytest.warns(RuntimeWarning, match='a matching pattern'):
        for method in set(TO_BE_IMPLEMENTED_METHODS).union(set(NOT_VALID_TOKENS)):
            _ = ot.unbalanced.sinkhorn(default, default, method=method, None, whatever)
```
There is no need to check for the output nor pass proper inputs. The only thing that you want to test is that given a method it goes to the right `if..else` branch.

Reply:
Done